### ISSUE TITLE
script.elementum.jackett stuck on version 0.0.1

### ISSUE THREAD

**Issue by antonsoroko** [ElementumOrg/repository.elementumorgold#2]
for some reason it stuck on version 0.0.1, while it already has version [v3.0.0](https://github.com/fugkco/script.elementum.jackett/releases/tag/v3.0.0)

https://github.com/ElementumOrg/repository.elementumorg/blob/5506e2731eb24c7f9b1082ab491bf0e2d85593a0/resources/repository.json#L24-L27

also, for reference - https://github.com/fugkco/script.elementum.jackett/issues/31

**Comment by elgatito** [comment]
Yes, that is because repository expects a new version in `addon.xml`.

That is the way how Kodi identifies a newer version. It does not look on zip file name, but reads `addon.xml` to parse version.

@fugkco  Would be great if you could update a version with `addon.xml` changes to list actual version name.

**Comment by fugkco** [comment]
I'm not sure I follow. The release process does update version attribute inside addon.xml:

```
(apk) [root:/tmp]# wget https://github.com/fugkco/script.elementum.jackett/releases/download/v3.0.0/script.elementum.jackett.zip
Connecting to github.com (140.82.121.4:443)
Connecting to objects.githubusercontent.com (185.199.110.133:443)
saving to 'script.elementum.jackett.zip'
script.elementum.jac 100% |************************************************************************************************************************************************|  695k  0:00:00 ETA
'script.elementum.jackett.zip' saved

(apk) [root:/tmp]# unzip script.elementum.jackett.zip
Archive:  script.elementum.jackett.zip
   creating: script.elementum.jackett/
  inflating: script.elementum.jackett/LICENSE
   creating: script.elementum.jackett/src/
  inflating: script.elementum.jackett/src/torrent.py
...
  inflating: script.elementum.jackett/resources/images/icon.png
  inflating: script.elementum.jackett/addon.xml

(apk) [root:/tmp]# grep version script.elementum.jackett/addon.xml
<?xml version="1.0" ?><!-- note version attribute gets changed on the fly using ./scripts/update-version.py --><addon id="script.elementum.jackett" name="Elementum [COLOR FF0291F7]Jackett[/COLOR]" version="3.0.0" provider-name="fugkco">
            <import addon="xbmc.python" version="3.0.0"/>
```

Or do you mean the filename of the release has to have the version in the file name?

**Comment by elgatito** [comment]
@fugkco Repository is reading directly `addon.xml` from the repo: https://github.com/fugkco/script.elementum.jackett/blob/master/addon.xml

To see what is the latest version, because it does not know about releases and zip files (that would be a huge overhead to read zip files for each version check).

**Comment by vasilky3** [comment]
@elgatito , But I can't see addon.xml in burst repo.
And it's updating
How does it work?

**Comment by elgatito** [comment]
@vasilky3 How you don't see it? - https://github.com/elgatito/script.elementum.burst/blob/master/addon.xml

**Comment by vasilky3** [comment]
@elgatito, sorry been watching from phone. forgot to press "show all"

**Comment by fugkco** [comment]
@elgatito where does it check the version? If you don't mind, I can change it to be github release version and raise a PR

**Comment by elgatito** [comment]
@fugkco https://github.com/ElementumOrg/repository.elementumorg/blob/master/lib/repository.py#L124
It not probably even the check of the version. It just takes `addon.xml` for each addon and make a common `addons.xml`

**Comment by antonsoroko** [comment]
@fugkco 

i think it makes more sense to commit your updated `addon.xml` (after it was altered by `scripts/update-version.py`) in 
https://github.com/fugkco/script.elementum.jackett/blob/master/.github/workflows/publish.yml
so your git repo will always have correct data.

rather than making a hack  (specific for your approach) in `repository.py` that would read your `addon.xml`, read latest tag, replace version in XML with that tag (basically a copy-paste of your approach from `scripts/update-version.py`).

although, if you add logic this as option (let's say `"replace_addon_version_with_latest_tag"`) - then maybe it is ok.

```
  {
    "id": "script.elementum.jackett",
    "username": "fugkco",
    "assets": {
      "zip": "https://github.com/{username}/{repository}/releases/download/v{version}/{id}.zip"
    },
    "replace_addon_version_with_latest_tag": true
  },
  ```
